### PR TITLE
LEARNER-3972 Test - Android - Automate Whats New Screen

### PR DESCRIPTION
**Pull request by nedo**
* Verify that Whats New screen has following contents,
** "Version New Features" as heading
** Cross-icon
** Gallery of the New features, having following contents in each page,
*** New Feature's Icon
*** New Feature's Title
*** New Feature's Details
*** "Done" button at end
* Verify that following values/strings/labels are correct,
** App Version
** Feature Name/Title
** Feature Details
* Verify that on tapping cross-icon will load Main Dashboard screen
* Verify user should be able to navigate between features
* Verify on tapping "Done" will load Main Dashboard screen
* Verify after re-login with same user "Whats New" screen will not be visible

### CODE REVIEW

**Comment by nedo** [comment]
@raeeschachar PR is ready for review

cc @staubina

File: android/tests/test_android_whats_new.py [hunk 1]
```diff
@@ -32,21 +41,69 @@ def test_validate_ui_elements(self, set_capabilities, setup_logging):
...
+
+            assert android_whats_new_page.get_title_textview().text == strings.WHATS_NEW_ANDROID_SCREEN_TITLE
+            assert android_whats_new_page.get_cross_icon()
+            assert android_whats_new_page.get_main_image().text == strings.BLANK_FIELD
+            assert android_whats_new_page.get_feature_title_textview().text == strings.WHATS_NEW_FEATURE_TITLE
+            assert android_whats_new_page.get_feature_details().text == strings.WHATS_NEW_FEATURE_DETAILS
+
+        else:
+            setup_logging.info('validate_ui_elements is not needed')
+            assert True
```
Comment (raeeschachar) [inline android/tests/test_android_whats_new.py]: `assert True` won't be needed, we can simply remove it

File: android/tests/test_android_whats_new.py [hunk 2]
```diff
@@ -32,21 +41,69 @@ def test_validate_ui_elements(self, set_capabilities, setup_logging):
...
 
-        assert android_whats_new_page.get_title_textview().text == strings.WHATS_NEW_ANDROID_SCREEN_TITLE
-        assert android_whats_new_page.get_cross_icon()
-        assert android_whats_new_page.get_main_image().text == strings.BLANK_FIELD
-        assert android_whats_new_page.get_feature_title_textview().text == strings.WHATS_NEW_FEATURE_TITLE
-        assert android_whats_new_page.get_feature_details().text == strings.WHATS_NEW_FEATURE_DETAILS
-        assert android_whats_new_page.get_done_button().text == strings.WHATS_NEW_DONE
+        else:
+            setup_logging.info('navigate_features is not needed')
+            assert True
```
Comment (raeeschachar) [inline android/tests/test_android_whats_new.py]: Same here, we can remove this line

File: android/tests/test_android_whats_new.py [hunk 3]
```diff
@@ -32,21 +41,69 @@ def test_validate_ui_elements(self, set_capabilities, setup_logging):
...
-        android_whats_new_page = AndroidWhatsNew(set_capabilities, setup_logging)
-        assert android_whats_new_page.exit_features() == Globals.MAIN_DASHBOARD_ACTIVITY_NAME
+        global_contents = Globals(setup_logging)
+        if global_contents.is_first_time:
+            android_whats_new_page = AndroidWhatsNew(set_capabilities, setup_logging)
+            assert android_whats_new_page.exit_features() == Globals.MAIN_DASHBOARD_ACTIVITY_NAME
+
+        else:
+            setup_logging.info('close_features is not needed')
+            assert True
```
Comment (raeeschachar) [inline android/tests/test_android_whats_new.py]: Same here

File: android/tests/test_android_whats_new.py (hunk 1 above)
Comment (nedo) [reply android/tests/test_android_whats_new.py]: done

File: android/tests/test_android_whats_new.py (hunk 2 above)
Comment (nedo) [reply android/tests/test_android_whats_new.py]: done

File: android/tests/test_android_whats_new.py (hunk 3 above)
Comment (nedo) [reply android/tests/test_android_whats_new.py]: done

**Comment by nedo** [comment]
@staubina PR is ready for Review.

Regards,
Naveed

**Pull request merged**